resources/datahub/resource_aws_datahub.go

```diff
@@ -103,9 +109,13 @@ func (r *awsDatahubResource) Read(ctx context.Context, req resource.ReadRequest,
 		return
 	}
 
+	clusterName := state.Name.ValueString()
+	if len(clusterName) == 0 {
+		clusterName = state.ID.ValueString()
+	}
```

Comment:
in what scenario could this happen?

Reply:
In the import scenario. :) Import uses the read to collect data for the resource. However during import it is not in the state, so the name will be null, we will only have the ID. You need to set the ID during import:
`terraform import cdp_environments_azure_environment.my-azure-environment-flexible "crn:cdp:environments:us-west-1:9d74eee4-1cad-45d7-b645-7ccf9edbb73d:environment:bda11374-3de4-4733-a266-2dc94a6672fa"`